CRM/Contribute/BAO/Contribution.php

```diff
@@ -3858,7 +3858,86 @@ public static function recordAdditionalPayment($contributionId, $trxnsData, $pay
     $params['skipLineItem'] = TRUE;
     $trxnsData['trxn_date'] = !empty($trxnsData['trxn_date']) ? $trxnsData['trxn_date'] : date('YmdHis');
     $arAccountId = CRM_Contribute_PseudoConstant::getRelationalFinancialAccount($contributionDAO->financial_type_id, 'Accounts Receivable Account is');
-    if ($paymentType == 'owed') {
+    $fiancialItemPaidStatusID = CRM_Core_PseudoConstant::getKey('CRM_Financial_DAO_FinancialItem', 'status_id', 'Paid');
+    if ($paymentType == 'overpaid') {
+      $params['partial_payment_total'] = $contributionDAO->total_amount + CRM_Utils_Array::value('total_amount', $trxnsData, 0.00);
+      $params['partial_amount_pay'] = CRM_Utils_Array::value('total_amount', $trxnsData, 0.00);
+      $params['pan_truncation'] = CRM_Utils_Array::value('pan_truncation', $trxnsData);
+      $params['card_type_id'] = CRM_Utils_Array::value('card_type_id', $trxnsData);
+      $params['check_number'] = CRM_Utils_Array::value('check_number', $trxnsData);
+      $params['financial_type_id'] = $contributionDAO->financial_type_id;
+      $trxnsData['net_amount'] = !empty($trxnsData['net_amount']) ? $trxnsData['net_amount'] : $trxnsData['total_amount'];
+      $trxnsData['to_financial_account_id'] = CRM_Financial_BAO_FinancialTypeAccount::getInstrumentFinancialAccount($trxnsData['payment_instrument_id']);
+
+      // add financial trxn for the overpaid amount
+      $financialTrxn = CRM_Contribute_BAO_Contribution::recordFinancialAccounts($params, $trxnsData);
+
+      $lineItems = CRM_Price_BAO_LineItem::getLineItemsByContributionID($contributionDAO->id);
+      $lineItemID = NULL;
+      // if line item found then lookout for the one which is configured  with contribution
+      // quick config price field, whose line_total will be updated with overpaid amount.
+      if (!empty($lineItems)) {
+        // find line item params which is used later to store the over paid amount
+        $priceSetId = CRM_Core_DAO::getFieldValue('CRM_Price_DAO_PriceSet', 'default_contribution_amount', 'id', 'name');
+        foreach ($lineItems as $id => $lineItem) {
+          if ($lineItem['price_set_id'] == $priceSetId) {
+            $lineItemID = $id;
+            $lineItemParams = array(
+              'id' => $id,
+               // update both line_total and unit_price with additional overpaid amount
+              'line_total' => $lineItem['line_total'] + $trxnsData['total_amount'],
+              'unit_price' => $lineItem['unit_price'] + $trxnsData['total_amount'],
+            );
+            civicrm_api3('LineItem', 'create', $lineItemParams);
+            break;
+          }
+        }
+      }
+      // If lineitem not found then create a new line item referring to default
```

Comment:
I don't understand this. How can there ever be a contribution without a line item? If there is, it is an error condition, and we shouldn't try to correct by inserting a line item.